vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -580,10 +580,13 @@ void output_example(vw& all, ccb& c, multi_ex& ec_seq)
   size_t num_features = 0;
   float loss = 0.;
 
-  // Should this be done for shared, action and slot?
   for (auto* ec : ec_seq)
   {
-    num_features += ec->num_features;
+    // Only features in action examples should be counted. This is because we
+    // merged the slot into the shared example and then later down the stack the
+    // shared example is merged into each action.
+    if (ec->l.conditional_contextual_bandit.type == CCB::example_type::action)
+    { num_features += ec->get_num_features(); }
```

Comment:
does that mean we never cound the slot/shared example features?

Reply:
The feature counting in CCB is still not quite right, this is a step in the right direction but not enough. It's incorrect to count shared features once, really this needs to be counted for each iteration. This is just a step in the right direction (I think)

Reply:
Would you prefer if this change was omitted until it can be fixed properly?

Reply:
would
```if (type == ::action)
         num_features += ec->get_num_features() + ec->get_num_features() * num_shared_features
```

be wrong?

I don't have a strong preference if things are being done incrementally that's fine with me, maybe add a TODO?

Reply:
I don't think thats right either. I think the tricky thing here is that you need to accommodate for the fact that slot features are merged into shared but then also count taking into account how `cb_adf` merges shared features into each action.

I'll add a todo because I want to come back to it.